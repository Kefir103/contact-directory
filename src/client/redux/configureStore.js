import { createStore, applyMiddleware, compose } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension/index';
import { rootReducer } from './reducers/rootReducer';
import thunk from 'redux-thunk';

let store;

export default function configureStore(initialState) {
    if (store) {
        return store;
    }

    store =
        process.env.NODE_ENV === 'production'
            ? createStore(rootReducer, initialState, compose(applyMiddleware(thunk)))
            : createStore(rootReducer, initialState, composeWithDevTools(applyMiddleware(thunk)));

    return store;
}
