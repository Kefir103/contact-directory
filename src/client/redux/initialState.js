export const initialState = {
    filter: {
        sortingMap: new Map([]),
        filterText: '',
        filterFields: [],
        currentPage: 1,
    },
    data: {
        currentElement: {
            elementInfo: undefined,
            elementIndex: -1,
        },
        elements: [],
        currentElements: [],
        filteredElements: undefined,
        inputContainer: {
            isAddButtonDisabled: false,
            isFormOpen: false,
            inputElement: {
                id: 0,
                firstName: '',
                lastName: '',
                email: '',
                phone: '',
            },
            validInputs: {
                id: false,
                firstName: false,
                lastName: false,
                email: false,
                phone: false,
            },
            addingStatus: {
                addingStatusText: '',
                isError: false,
            },
        },
    },
    appStatus: {
        isLoading: false,
        error: undefined,
    },
};
