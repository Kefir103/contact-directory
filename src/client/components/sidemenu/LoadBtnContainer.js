import React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { loadData } from '../../redux/actions/dataActions';

const LoadBtnContainer = (props) => {
    const urlSmallData =
        'http://www.filltext.com/?rows=32&id={number|1000}&firstName={firstName}&lastName={lastName}&email={email}&phone={phone|(xxx)xxx-xx-xx}&address={addressObject}&description={lorem|32}';
    const urlBigData =
        'http://www.filltext.com/?rows=1000&id={number|1000}&firstName={firstName}&delay=3&lastName={lastName}&email={email}&phone={phone|(xxx)xxx-xx-xx}&address={addressObject}&description={lorem|32}';

    return (
        <p className={'load-btn-container'}>
            <button
                className={'load-btn'}
                onClick={() => props.actions.loadData(urlSmallData)}>
                Загрузить малый объем данных
            </button>
            <button
                className={'load-btn'}
                onClick={() => props.actions.loadData(urlBigData)}>
                Загрузить большой объем данных
            </button>
        </p>
    );
};

const mapDispatchToProps = (dispatch) => {
    return {
        actions: bindActionCreators(
            {
                loadData,
            },
            dispatch
        ),
    };
};

export default connect(null, mapDispatchToProps)(LoadBtnContainer);
