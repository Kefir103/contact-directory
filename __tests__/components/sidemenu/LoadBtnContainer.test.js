import React from 'react';
import { shallow, render } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import LoadBtnContainer from '../../../src/client/components/sidemenu/LoadBtnContainer';
import { initialState } from '../../../src/client/redux/initialState';
import thunk from 'redux-thunk';
import fetchMock from 'fetch-mock-jest';

const mockStore = configureMockStore([thunk]);

describe('<LoadBtnContainer />', () => {
    let store;
    beforeEach(() => {
        store = mockStore({ ...initialState });
    });
    afterEach(() => {
        fetchMock.restore();
    });
    test('should render correctly', () => {
        const wrapper = shallow(<LoadBtnContainer store={store} />);
        expect(wrapper.dive().dive()).toMatchSnapshot();
    });
});
