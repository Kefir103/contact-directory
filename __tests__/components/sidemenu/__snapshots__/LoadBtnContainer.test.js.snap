// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadBtnContainer /> should render correctly 1`] = `
<div
  className="load-btn-container"
>
  <button
    className="load-btn"
    onClick={[Function]}
  >
    Загрузить малый объем данных
  </button>
  <button
    className="load-btn"
    onClick={[Function]}
  >
    Загрузить большой объем данных
  </button>
</div>
`;
